universal-login-sdk/lib/core/services/ExecutionFactory.ts

```diff
@@ -24,23 +24,38 @@ export class ExecutionFactory {
     ensureNotNull(result.status.messageHash, MissingMessageHash);
     const {messageHash, totalCollected, required} = result.status;
     const waitToBeMined = totalCollected >= required ? this.createWaitToBeMined(messageHash) : async () => result.status;
-    const waitToBePending = async () => {
-      throw Error('Not implemented');
-    };
+    const waitToExecutionStart = totalCollected >= required ? this.createExecutionStart(messageHash) : async () => result.status;
     return {
       messageStatus: result.status,
       waitToBeMined,
-      waitToBePending
+      waitToExecutionStart
     };
   }
 
   private isExecuted(messageStatus: MessageStatus) {
-    return !!messageStatus.transactionHash || !!messageStatus.error;
+    return messageStatus.state === 'Error' || messageStatus.state === 'Success';
+  }
+
+  private isStarted(messageStatus: MessageStatus) {
+    return ['Pending', 'Success', 'Error'].includes(messageStatus.state);
+  }
+
+  private createGetStatus(messageHash: string) {
+    return async () => this.relayerApi.getStatus(messageHash);
+  }
```

Comment:
private getStatus(messageHash: string) {
    return this.relayerApi.getStatus(messageHash);
  }